opennms-base-assembly/src/main/filtered/bin/find-java.sh

```diff
@@ -50,11 +50,14 @@ get_java_version_string() {
 get_real_path() {
 	file_to_find="$1"
 	readlink="$(command -v readlink 2>/dev/null)"
-	if [ -n "$readlink" ] && [ -n "$file_to_find" ] && [ -e "$file_to_find" ]; then
-		"$readlink" "$file_to_find" || echo "$file_to_find"
-	else
-		echo "$file_to_find"
+	if [ -n "$readlink" ]; then
+		__new_filename="$("$readlink" "$file_to_find" || :)"
+		if [ -n "$__new_filename" ] && [ -e "$__new_filename" ]; then
```

Comment:
You can use `-h` or `-L` in tests to determine if the file is a link and use this to end your recursion.

Reply:
on Mac OS X `readlink` only supports the `-n` option

Reply:
I'm not talking about `readlink` but about `test`. Something like
```
if [ -h "$file_to_find" ]; then
    get_real_path "$("$readlink" "$file_to_find")
else
  echo "$file_to_find"
fi
```

Reply:
oh!  I gotcha, yeah, that makes sense